hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java

```diff
@@ -155,14 +156,16 @@ private void initNameNode(ExecutorDriver driver, String dnsName) throws Exceptio
         // is used as a mutex indicates whether or not any NameNode has ever been formatted.
         // The first NameNode to acquire the mutex and find that no NameNode has ever been
         // formatted, formats itself.  All others bootstrap.
```

Comment:
I don't think this logic should change. Just because we are backing up, we should still be sure to format the first NN and bootstrap the second. Actually recovering from backup should be part of our state machine. 

When formatting or bootstrapping the NN, we should then decide to do a backup. I see here you chose to create a backup of the formatted namenode only. What was the reasoning behind that?